builtin/logical/aws/client.go

```diff
@@ -124,22 +128,31 @@ func (b *backend) getRootConfigs(ctx context.Context, s logical.Storage, clientT
 		opts = append(opts, awsutil.WithEnvironmentCredentials(false), awsutil.WithSharedCredentials(false))
 	}
 
-	// at this point, in the IAM case, regions contains nothing, and endpoints contains iam_endpoint, if it was set.
+	// at this point, in the IAM case, regions contains config.Region, if it was set.
+	//                                 endpoints contains iam_endpoint, if it was set.
 	// in the sts case, regions contains sts_region, if it was set, then the sts_fallback_regions in order, if they were set.
 	//                  endpoints contains sts_endpint, if it wa set, then sts_fallback_endpoints in order, if they were set.
 
+	logger.Info("pre", "region", fmt.Sprintf("%+v", regions), "endpoint", fmt.Sprintf("%+v", endpoints))
+
 	// case in which nothing was supplied
 	if len(regions) == 0 {
 		// fallback region is in descending order, AWS_REGION, or AWS_DEFAULT_REGION, or us-east-1
 		regions = append(regions, fallbackRegion)
+	}
 
-		// we also need to set the endpoint based on this region (since we need matched length arrays)
-		if len(endpoints) == 0 {
-			switch clientType {
-			case "sts":
-				endpoints = append(endpoints, matchingSTSEndpoint(fallbackRegion))
-			case "iam":
-				endpoints = append(endpoints, "https://iam.amazonaws.com") // see https://docs.aws.amazon.com/general/latest/gr/iam-service.html
+	if len(endpoints) == 0 {
```

Comment:
It is possible for no endpoint to be supplied even if a region was - in particular in the IAM case, which expects no endpoint unless a third-party reimplementer of IAM is being used.